libarchive/archive_match.c

```diff
@@ -781,7 +807,9 @@ static int
 match_path_inclusion(struct archive_match *a, struct match *m,
     int mbs, const void *pn)
 {
-	int flag = PATHMATCH_NO_ANCHOR_END;
+	int flag = a->recursive_include ?
+		PATHMATCH_NO_ANCHOR_END : /* Prefix match is good enough. */
+		0; /* Full match required. */
```

Comment:
Split the comment, i.e. `/* Recursive operation requires only a prefix match. */`

Reply:
You mean like this?
```
	/* Recursive operation requires only a prefix match. */
	int flag = a->recursive_include ?
		PATHMATCH_NO_ANCHOR_END :
		0;
```

That's what I am using now.